woocommerce/google-listings-and-ads
# Add the Google account reconnection UI and flow

### Changes proposed in this Pull Request:

Implemented the rest of #486 and based on #834.

Please refer to the **Google account reconnection flow** section in the #486 description.

- Add a new shared component AccountCard for the new design
- Add selector/resolver/action/reducer to get Google access status from API and to disconnect Google account
- Implement the Google account reconnection UI and flow
- Redirect to the reconnection page from the Setting page if the Google account is disconnected

#### 💡 Additional note

Although the improvement of onboarding process design is still under discussion, the new design for the account card UI has already been included in this PR, so the component was implemented in a way that will continue to be developed in the future. 

### Screenshots:

#### 🎥 . Reconnect to a Google account with access permissions
https://user-images.githubusercontent.com/17420811/123742036-a571c200-d8dd-11eb-8e63-c924dcb5971f.mp4

#### 🎥 . Reconnect to a Google account that does not have access permissions
https://user-images.githubusercontent.com/17420811/123742249-ed90e480-d8dd-11eb-8c89-0315d27f32ee.mp4

### Detailed test instructions:

1. Disconnect Google account from the Connection Test page
2. Go to the dashboard page
3. When receiving any 401 response error or detecting the Google account being disconnected, the browser should be redirected to the reconnection page `/wp-admin/admin.php?page=wc-admin&subpath=%2Freconnect-accounts&path=%2Fgoogle%2Fsettings`
4. Proceed with next steps according to the **Google account reconnection flow** section in the #486 description
5. See if the implementation of reconnection flow is working properly as that description

####❗ An edge case

When browsing to the Product Feed page with a disconnected Google account, the `/wc/gla/mc/product-statistics` API returns 400 response error, so it'd not be redirected to the reconnection page. I was thinking if users would be redirected to reconnect by other pages eventually, maybe we could leave it as is.

### Changelog entry

> Add - Google account reconnection UI and flow.
